src/nunavut/jinja/lang/py.py

```diff
@@ -53,3 +59,86 @@ def filter_to_template_unique_name(env: Environment, base_token: str) -> str:
         be unique within the file generated by the current template.
     """
     return env.globals['_unique_name_generator']('py', base_token, '_', '_')  # type: ignore
+
+
+PYTHON_RESERVED_IDENTIFIERS = frozenset(map(str, list(keyword.kwlist) + dir(builtins)))  # type: typing.FrozenSet[str]
+
+
+def filter_id(instance: typing.Any, stropping_suffix: str = '_', encoding_prefix: str = 'ZX') -> str:
+    """
+    Filter that produces a valid Python identifier for a given object. The encoding may not
+    be reversable.
+
+    Example::
+
+        {{ "I like python" | py.id }}
+        {{ "&because" | py.id }}
+        {{ "if" | py.id }}
+        {{ "if" | py.id('stropped_') }}
+
+    Results Example::
+
+       I_like_python
+       ZX0038because
+       _if
```

Comment:
```suggestion
       if_
```

The code and the tests have been updated but not the docs.